deepchecks/vision/utils/transformations.py

```diff
@@ -8,36 +8,74 @@
 from deepchecks.core.errors import DeepchecksNotSupportedError, DeepchecksValueError
 
 
-def add_augmentation_in_start(aug, transforms):
-    """Return new object which contains given augmentation in the start and then the rest of transforms."""
-    transform_type = get_transform_type(transforms)
-    if transform_type == 'albumentations':
+class ImgaugTransformations:
+
+    @classmethod
+    def add_augmentation_in_start(cls, aug, transforms):
+        if not isinstance(aug, imgaug.augmenters.Augmenter):
+            raise DeepchecksValueError(f'Transforms is of type imgaug, can\'t add to it type {type(aug)}')
+        return imgaug.augmenters.Sequential([aug, transforms])
+
+    @classmethod
+    def get_test_transformation(cls):
+        return imgaug.augmenters.Rotate(rotate=(20, 30))
+
+    @classmethod
+    def get_robustness_augmentations(cls, data_dim):
+        return [
+            imgaug.augmenters.MultiplyHueAndSaturation()
+        ]
+
+
+class AlbumentationsTransformations:
+
+    @classmethod
+    def add_augmentation_in_start(cls, aug, transforms):
         if not isinstance(aug, (albumentations.Compose, albumentations.BasicTransform)):
             raise DeepchecksValueError(f'Transforms is of type albumentations, can\'t add to it type {type(aug)}')
         # Albumentations compose contains preprocessors and another metadata needed, so we can't just create a new one,
         # so we need to copy it.
         album_compose = copy(transforms)
         album_compose.transforms = [aug, *album_compose.transforms]
         return album_compose
-    elif transform_type == 'imgaug':
-        if not isinstance(aug, imgaug.augmenters.Augmenter):
-            raise DeepchecksValueError(f'Transforms is of type imgaug, can\'t add to it type {type(aug)}')
-        return imgaug.augmenters.Sequential([aug, transforms])
-    else:
-        raise DeepchecksNotSupportedError(f'Not implemented for type {transform_type}')
+
+    @classmethod
+    def get_test_transformation(cls):
+        return imgaug.augmenters.Rotate(rotate=(20, 30))
+
+    @classmethod
+    def get_robustness_augmentations(cls, data_dim):
+        augmentations = [
```

Comment:
So, are the actual augmentations hard coded here? Can't the user input them in the constructor?
By the way- I'd assert that the given list of robustness- augmentations is the same type of the inner-attribute given by the user in `__getattr__`

Reply:
This are default, user can pass his own. (It's not here, it's in the check)
And we do asserting, we return the appropriate type by the type the user have in his field